### PR TITLE
Add function `spherical_to_cartesian` to `features`

### LINKED ISSUE #4979: Add `spherical_transform` function
### Describe the feature you would like to be added.

`SphericalTransform` will convert (r,phi,theta) coordinates to (x,y,z) coordinates and back again. The equation that is used is $x = r \sin(phi) \cos(theta)$ , $y = r \sin(phi) \sin(theta)$ , $z = r \cos(phi)$ .

_Originally posted by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4965#discussion_r1341733386_

### Links to VTK Documentation, Examples, or Class Definitions.

https://vtk.org/doc/nightly/html/classvtkSphericalTransform.html#details

### Pseudocode or Screenshots

_No response_

### PR DESCRIPTION
**Pull request by denizkirbiyik**
Added a function that converts spherical coordinates to 3D cartesian coordinates using the following formulas:
x = r*sin(theta)*cos(phi)
y = r*sin(theta)*sin(phi)
z = r*cos(theta)

resolves #4979

### CODE REVIEW

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

**Comment by tkoyama010** [comment]
@denizkirbiyik 
Thank you for the PR. This is a very nice addition. We need to add testing to this. Mention me if you need help.

File: pyvista/core/utilities/features.py [hunk 1]
```diff
@@ -247,6 +247,34 @@ def cartesian_to_spherical(x, y, z):
...
+def spherical_to_cartesian(r, theta, phi):
+    """Convert Spherical coordinates to 3D Cartesian coordinates.
+
+    Parameters
+    ----------
+    r : numpy.ndarray
+        Radial distance.
+
+    theta : numpy.ndarray
+        Angle (radians) with respect to the polar axis. Also known
```
Comment (MatthewFlamm) [inline pyvista/core/utilities/features.py:259]: vtk defines theta and phi the other way around.  I think we should align to how vtk operates.

See https://vtk.org/doc/nightly/html/classvtkSphericalTransform.html#details

#4965 and `pyvista.Sphere` are already consistent with this vtk operation.

File: pyvista/core/utilities/features.py [hunk 2]
```diff
@@ -247,6 +247,34 @@ def cartesian_to_spherical(x, y, z):
...
+        as polar angle.
+
+    theta : numpy.ndarray
+        Angle (radians) of rotation from the initial meridian plane.
+        Also known as azimuthal angle.
+
+    Returns
+    -------
+    x, y, z : numpy.ndarray
```
Comment (tkoyama010) [inline pyvista/core/utilities/features.py:268]:
```suggestion
    numpy.ndarray, numpy.ndarray, numpy.ndarray
```

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

File: tests/core/test_utilities.py [hunk 3]
```diff
@@ -747,15 +747,17 @@ def test_copy_vtk_array():
...
     x, y, z = points.T
     r, theta, phi = pv.cartesian_to_spherical(x, y, z)
-    assert np.allclose(polar2cart(r, theta, phi), points)
+    assert np.allclose(pv.spherical_to_cartesian(r, theta, phi), points)
+
+
+def test_spherical_to_cartesian():
+    points = np.random.random((1000, 3))
+    r, theta, phi = points.T
+    x, y, z = pv.spherical_to_cartesian(r, theta, phi)
```
Comment (tkoyama010) [inline tests/core/test_utilities.py:759]:
```suggestion
    x, y, z = pv.spherical_to_cartesian(r, phi, theta)
```

File: tests/core/test_utilities.py [hunk 4]
```diff
@@ -747,15 +747,17 @@ def test_copy_vtk_array():
...
-    def polar2cart(r, theta, phi):
-        return np.vstack(
-            (r * np.sin(theta) * np.cos(phi), r * np.sin(theta) * np.sin(phi), r * np.cos(theta))
-        ).T
-
     points = np.random.random((1000, 3))
     x, y, z = points.T
     r, theta, phi = pv.cartesian_to_spherical(x, y, z)
-    assert np.allclose(polar2cart(r, theta, phi), points)
+    assert np.allclose(pv.spherical_to_cartesian(r, theta, phi), points)
```
Comment (tkoyama010) [inline tests/core/test_utilities.py:753]:
```suggestion
    assert np.allclose(pv.spherical_to_cartesian(r, phi, theta), points)
```

File: tests/core/test_utilities.py [hunk 5]
```diff
@@ -747,15 +747,17 @@ def test_copy_vtk_array():
...
     points = np.random.random((1000, 3))
     x, y, z = points.T
     r, theta, phi = pv.cartesian_to_spherical(x, y, z)
-    assert np.allclose(polar2cart(r, theta, phi), points)
+    assert np.allclose(pv.spherical_to_cartesian(r, phi, theta), points)
+
+
+def test_spherical_to_cartesian():
+    points = np.random.random((1000, 3))
+    r, theta, phi = points.T
```
Comment (tkoyama010) [inline tests/core/test_utilities.py:758]:
```suggestion
    r, phi, theta = points.T
```

**Review by tkoyama010 — DISMISSED** [review]
LGTM. Thanks for your contribution!

**Review by tkoyama010 — APPROVED** [review]

**Review by tkoyama010 — CHANGES_REQUESTED** [review]
~~Because the backward compatibility of `cartesian_to_spherical` needs to be addressed, block the merge until the review is complete.~~ _Edit: I'm sorry. I was mistaken._

**Review by tkoyama010 — APPROVED** [review]

**Review by tkoyama010 — CHANGES_REQUESTED** [review]
Please wait until #5014 is merged to organize PRs.

**Review by tkoyama010 — APPROVED** [review]

**Pull request merged**